### PR TITLE
Add ability to use all controllers round robin

### PR DESCRIPTION
**Pull request by cbickel**
This PR changes the nginx to use all controllers round robin.

This PR is blocked by:
- [x] #2624 
- [x] #2531 
- [x] #2697 

In addition it removes the cleanup of containers, called `controller` (without index)

@vvraskin

### CODE REVIEW

**Comment by ningyougang** [comment]
#2531 has been merged!

**Comment by cbickel** [comment]
PG3#1191 is 🔵

File: tests/src/test/scala/whisk/core/database/test/CacheConcurrencyTests.scala [hunk 1]
```diff
@@ -121,14 +124,24 @@ class CacheConcurrencyTests extends FlatSpec with WskTestHelpers with BeforeAndA
...
-        case (name, rr) =>
-          withClue(s"get after update: failed check for $name") {
-            rr.stdout should include("blue")
-            rr.stdout should not include ("red")
+      // All controllers should have the correct action
+      // As they are used round robin, we ask every controller for the action.
+      // We add a retry to tollarate a short interval to bring the controllers in sync.
+      retry(
+        {
+          (1 to WhiskProperties.getControllerHosts.split(",").length).foreach { _ =>
```
Comment (markusthoemmes) [inline tests/src/test/scala/whisk/core/database/test/CacheConcurrencyTests.scala]: aka

```scala
WhiskProperties.getControllerHosts.split(",").foreach { _ =>
```

?

**Comment by cbickel** [comment]
The following PR will be merged instead: #2888

**Pull request closed without merging**